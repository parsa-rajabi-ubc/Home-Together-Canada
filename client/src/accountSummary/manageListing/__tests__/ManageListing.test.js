/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.06.13
 *
 * @Description: Manage Listing Test
 *
 */
import React from 'react';
import renderer from  'react-test-renderer'
import ManageListing from "../ManageListing";

describe('ManageListing', () => {
    describe('Snapshot test', () => {
        it("should render correctly regardless of properties", () => {
            // given
            const props = {
                onSubmit: jest.fn(),
                listingData: [{
                    "id": 51,
                    "uid": 5,
                    "title": "Empty Room in Condo 7",
                    "shortDescription": "Looking for an easy going roommate",
                    "fullDescription": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
                    "monthlyCost": 600,
                    "numBed": 2,
                    "numBath": 1,
                    "petFriendly": true,
                    "smokeFriendly": false,
                    "utilitiesIncluded": true,
                    "addressLine1": "2271 Harvey Ave",
                    "city": "Kelowna",
                    "province": "BC",
                    "postalCode": "V1Y6H2",
                    "isClassified": false,
                    "createdAt": "2021-06-14T00:42:47.000Z",
                    "updatedAt": "2021-06-14T00:42:47.000Z",
                    "categoryName": "Members with Homes to Share",
                    "images": []
                }],
                activeTab: "Live"
            }

            // when
            const component = renderer.create(<ManageListing {...props} />).toJSON();

            // then
            expect(component).toMatchSnapshot();
        });
    })
})