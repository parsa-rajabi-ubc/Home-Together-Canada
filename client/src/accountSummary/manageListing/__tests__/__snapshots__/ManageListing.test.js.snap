// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageListing Snapshot test should render correctly regardless of properties 1`] = `
<div>
  <div>
    <div
      className="flex"
    >
      <section
        className="w-full"
      >
        <section
          className="card-container mt-0"
        >
          <div
            className="inline ml-6 align-middle "
          >
            <label
              className="font-semibold justify-between"
            >
               
              Empty Room in Condo 7
            </label>
            ,
            <label
              className="px-1"
            >
              $
              600
               
            </label>
            <section
              className="inline float-right pr-6 justify-end"
            >
              <time
                dateTime={"2021-06-14T00:42:47.000Z"}
              >
                Jun 14, 2021
              </time>
            </section>
          </div>
          <div
            className="ml-6 mt-4 flex"
          >
            <label
              className="w-full font-light"
            >
               
              Looking for an easy going roommate
               
            </label>
            <section
              className="flex pr-6 "
            >
              <span>
                <span
                  data-for="PetFriendly"
                  data-tip={true}
                >
                  <svg
                    className="text-blue-400 ml-1"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M298.06,224,448,277.55V496a16,16,0,0,1-16,16H368a16,16,0,0,1-16-16V384H192V496a16,16,0,0,1-16,16H112a16,16,0,0,1-16-16V282.09C58.84,268.84,32,233.66,32,192a32,32,0,0,1,64,0,32.06,32.06,0,0,0,32,32ZM544,112v32a64,64,0,0,1-64,64H448v35.58L320,197.87V48c0-14.25,17.22-21.39,27.31-11.31L374.59,64h53.63c10.91,0,23.75,7.92,28.62,17.69L464,96h64A16,16,0,0,1,544,112Zm-112,0a16,16,0,1,0-16,16A16,16,0,0,0,432,112Z"
                    />
                  </svg>
                </span>
                <div
                  className="__react_component_tooltip t00000000-0000-0000-0000-000000000000 place-top type-dark"
                  data-id="tooltip"
                  id="PetFriendly"
                >
                  <style
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "
  	.t00000000-0000-0000-0000-000000000000 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t00000000-0000-0000-0000-000000000000.place-top {
        margin-top: -10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-top::before {
        border-top: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-bottom {
        margin-top: 10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-left {
        margin-left: -10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-left::before {
        border-left: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-right {
        margin-left: 10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-right::before {
        border-right: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  ",
                      }
                    }
                  />
                  Pet Friendly
                </div>
              </span>
            </section>
          </div>
        </section>
      </section>
      <section
        className="my-8"
      >
        <svg
          color="#DB4437"
          fill="currentColor"
          height="40"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#DB4437",
            }
          }
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </section>
    </div>
    <ul
      className="flex m-auto w-1/4"
    >
      <li
        className="next-prev-text disabled"
      >
        <a
          aria-disabled="true"
          aria-label="Previous page"
          className="next-prev-link"
          onClick={[Function]}
          onKeyPress={[Function]}
          rel="prev"
          role="button"
          tabIndex="0"
        >
          Previous
        </a>
      </li>
      <li
        className="selected"
      >
        <a
          aria-current="page"
          aria-label="Page 1 is your current page"
          className="page-numbers current-page"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          1
        </a>
      </li>
      <li
        className="next-prev-text disabled"
      >
        <a
          aria-disabled="true"
          aria-label="Next page"
          className="next-prev-link"
          onClick={[Function]}
          onKeyPress={[Function]}
          rel="next"
          role="button"
          tabIndex="0"
        >
          Next
        </a>
      </li>
    </ul>
  </div>
</div>
`;
